examples/pubsub/tutorial_pubsub_mqtt_subscribe.c

```diff
@@ -0,0 +1,329 @@
+/* This work is licensed under a Creative Commons CCZero 1.0 Universal License.
+ * See http://creativecommons.org/publicdomain/zero/1.0/ for more information. */
+
+/**
+ * .. _pubsub-tutorial:
+ *
+ * Working with Publish/Subscribe
+ * ------------------------------
+ *
+ * Work in progress: This Tutorial will be continuously extended during the next
+ * PubSub batches. More details about the PubSub extension and corresponding
+ * open62541 API are located here: :ref:`pubsub`.
+ *
+ * Subscription Fields
+ * ^^^^^^^^^^^^^^^^^
+ * The PubSub MQTT subscribe example demonstrate the simplest way to subscribe
+ * informations from the information model over MQTT using the UADP (or later
+ * JSON) encoding. To receive information the subscribe functionality of mqtt is
+ * used. A periodical call to yield is necessary to update the mqtt stack.
+ *
+ * NOTE: For now, received message processing is skipped i.e. address space is not
+ * updated upon message reception. Instead a user callback function displays the received
+ * message.
+ *
+ * **Connection handling**
+ * PubSubConnections can be created and deleted on runtime. More details about
+ * the system preconfiguration and connection can be found in
+ * ``tutorial_pubsub_connection.c``.
+ */
+
+#include "open62541/server.h"
+#include "open62541/server_config_default.h"
+#include "ua_pubsub.h"
+#include "ua_network_pubsub_mqtt.h"
+#include "open62541/plugin/log_stdout.h"
+#include <signal.h>
+
+#define CONNECTION_NAME              "MQTT Subscriber Connection"
+#define TRANSPORT_PROFILE_URI        "http://opcfoundation.org/UA-Profile/Transport/pubsub-mqtt"
+#define MQTT_CLIENT_ID               "TESTCLIENTPUBSUBMQTT_SUB"
+#define CONNECTIONOPTION_NAME        "mqttClientId"
+#define TOPIC              			"customTopic"
+#define BROKER_ADDRESS_URL           "opc.mqtt://127.0.0.1:1883"
+
+
+//static UA_Boolean useJson = false;
+static UA_NodeId connectionIdentifier;
+static UA_NodeId readerGroupIdentifier;
+static UA_NodeId readerIdentifier;
+
+static UA_DataSetReaderConfig readerConfig;
+void subscriber_message_callback(UA_ByteString *encodedBuffer, UA_ByteString *topic);
+
+void subscriber_message_callback(UA_ByteString *encodedBuffer, UA_ByteString *topic) {
+	UA_LOG_INFO(UA_Log_Stdout, UA_LOGCATEGORY_USERLAND, "message received: topic=%s | size=%d.",
+	        topic->data, (int)encodedBuffer->length);
+	UA_LOG_INFO(UA_Log_Stdout, UA_LOGCATEGORY_USERLAND, "%s\n.", encodedBuffer->data);
```

Comment:
Can we extend this example to match the functionality of the non MQTT subscriber examples? It would be great to decode the received values (most common types) and print the 'real' content.